lib/elixir/lib/access.ex

```diff
@@ -31,83 +26,52 @@ defmodule Access do
+  accessing a non-existing key will _not raise_, but will instead return a `nil` value.
 
       iex> keywords = [a: 1, b: 2]
       iex> keywords[:c][:unknown]
       nil
 
-  Since `Access` is a behaviour, it can be implemented for key-value
-  data structures. The implementation should be added to the
-  module that defines the struct being accessed. `Access` requires the
-  key comparison to be implemented using the `===` operator.
-
-  ## Static lookups
-
-  The `Access` syntax (`data[key]`) cannot be used to access fields in
-  structs, since structs do not implement the `Access` behaviour by
-  default. It is also a design decision: the dynamic access lookup
-  is meant to be used for dynamic key-value structures, like maps
-  and keywords, and not by static ones like structs (where fields are
-  known and not dynamic).
-
-  Therefore Elixir provides a static lookup for struct fields and for atom
-  fields in maps. Imagine a struct named `User` with a `:name` field.
-  The following would raise:
+  Structs do not by default implement Access behaviour and so by default cannot use this
+  notation (see: [Structs and Dot Syntax](#module-structs-and-dot-syntax)) but
+  Elixir allows the Access behaviour to be [implemented](#module-implementing-the-behaviour)
+  for user-defined struct types after which, bracket syntax may be used.
 
-      user = %User{name: "John"}
-      user[:name]
-      # ** (UndefinedFunctionError) undefined function User.fetch/2 (User does not implement the Access behaviour)
-
-  Structs instead use the `user.name` syntax to access fields:
-
-      user.name
-      #=> "John"
+  ## Access and `Kernel`
 
-  The same `user.name` syntax can also be used by `Kernel.put_in/2`
-  for updating structs fields:
-
-      put_in user.name, "Mary"
-      #=> %User{name: "Mary"}
-
-  Differently from `user[:name]`, `user.name` is not extensible via
-  a behaviour and is restricted only to structs and atom keys in maps.
-
-  As mentioned above, this works for atom keys in maps as well. Refer to the
-  `Map` module for more information on this.
+  The `Kernel` module provides a small suite of "accessor" functions (such as
+  `Kernel.get_in/2`, `Kernel.put_in/3`, `Kernel.update_in/3`,
+  `Kernel.get_and_update_in/3`) to make reading and writing _nested_
+  values inside data structures more convenient. The implementation of these functions
+  rely heavily on the core Access functions.
 
-  Summing up:
+  ### Access Accessor Helpers
 
-    * `user[:name]` is used by dynamic structures, is extensible and
-      does not raise on missing keys
-    * `user.name` is used by static structures, it is not extensible
-      and it will raise on missing keys
+  In conjunction with the accessor functions in `Kernel`, Access provides helper
+  functions for traversing other structures, like tuples and lists.
 
-  ## Accessors
+  For example, `Access.all/0` returns a function that meets the requirements of
```

Comment:
This was important for me to know as it unveiled exactly how these `Access` functions could be used in `Kernel.put_in/3` (and friends).